src/analysis/special_functions/arsinh.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2020 James Arthur. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: James Arthur, Chris Hughes, Shing Tak Lam.
+-/
+import analysis.special_functions.trigonometric
+import data.real.basic
+import analysis.special_functions.pow
+
+open real
+local attribute [pp_nodot] real.log
+noncomputable theory
+
+/-!
+# Inverse of the sinh function
+
+In this file we prove that sinh is bijective and hence has an
+inverse, arsinh.
+
+## Main Results
+
+- `sinm_injective`: The proof that `sinm` is injective
+- `sinm_surjective`: The proof that `sinm` is surjective
+- `sinm_bijective`: The proof `sinm` is bijective
+- `arsinh`: The inverse function of `sinm`
+
+## Tags
+
+arsinh, sinh injective, sinh bijective, sinh surjective
+-/
+
+
+/-- `arsinh` is defined using a logarithm-/
+def arsinh (x : ℝ) := log (x + sqrt(1 + x^2))
+
+/-- `sinh` is injective, `∀ a b, sinh a = sinh b → a = b` -/
+lemma sinh_injective : function.injective sinh := sinh_strict_mono.injective
+
+private lemma aux_lemma (x : ℝ) : 1/(x + (1 + x^2).sqrt) = - x + (1 + x^2).sqrt :=
```

Comment:
You're sometimes using dot notation for `sqrt`, but sometimes you aren't. It might be nice to unify it.

Reply:
Right, I'll sort that out. Mainly because I copied some stuff from lean goal

Reply:
Done!